File: optimizer/src/optimizer/models/staff_unavailability.py
"""スタッフ希望休モデル — shared/types/staff-unavailability.ts に対応"""

from pydantic import BaseModel

from .common import UnavailableSlot


class StaffUnavailability(BaseModel):
    staff_id: str
    week_start_date: str  # "YYYY-MM-DD"
    unavailable_slots: list[UnavailableSlot] = []

File: shared/types/staff-unavailability.ts
import { Timestamp } from 'firebase-admin/firestore';
import { UnavailableSlot } from './common';

/** スタッフ希望休 */
export interface StaffUnavailability {
  id: string;
  staff_id: string;
  week_start_date: Timestamp;
  unavailable_slots: UnavailableSlot[];
  notes?: string;
  submitted_at: Timestamp;
}
